### ISSUE TITLE
Assembly VentrixSyncDisks.dll is not registered in il2cpp

### ISSUE THREAD

**Issue by ErraticFox** [Arsonide/VentVigilante#1]
It looks like the mod for the most part loads except it throws that error.
```
[Info   :   BepInEx] Loading [VentrixSyncDisks 0.8.0]
[Info   :VentrixSyncDisks] Plugin VentrixSyncDisks is loaded!
[Info   :VentrixSyncDisks] Plugin VentrixSyncDisks is patched!
[Info   :Il2CppInterop] Registered mono type VentrixSyncDisks.Implementation.Pooling.BasePoolObject in il2cpp domain
[Error  :Il2CppInterop] Assembly VentrixSyncDisks.dll is not registered in il2cpp
[Info   :Il2CppInterop] Registered mono type VentrixSyncDisks.Implementation.Mapping.DuctMarker in il2cpp domain
[Info   :Il2CppInterop] Registered mono type VentrixSyncDisks.Implementation.Mapping.EcholocationPulse in il2cpp domain
[Info   :Il2CppInterop] Registered mono type VentrixSyncDisks.Implementation.Snooping.SnoopHighlighter in il2cpp domain
[Info   :Il2CppInterop] Registered mono type VentrixSyncDisks.Implementation.Common.Timer in il2cpp domain
[Info   :VentrixSyncDisks] Plugin VentrixSyncDisks has added custom types!
```

**Comment by Self-Replicating-Moron** [comment]
+1 on this, it works perfectly fine for me up until I have to load a save, then , if I have any of the sync disks installed it wipes my inventory and installed syncs, sometimes restarting the whole world.

**Comment by Self-Replicating-Moron** [comment]
I have the feeling this wont get fixed, author hasn't been active since last year, rip

**Comment by ErraticFox** [comment]
> I have the feeling this wont get fixed, author hasn't been active since last year, rip

I do a little dabbling in this stuff, I'll look at it later when I'm not sick.

**Comment by Self-Replicating-Moron** [comment]
ah really, thankyou so much for offering, i cant code for shit, get well soon

although dont waste your time unless youre bored and want something to do, my enjoyment from the game has died for the time being, the whole game needs an update, even with mods everything gets so repetitive

**Comment by ErraticFox** [comment]
> ah really, thankyou so much for offering, i cant code for shit, get well soon
> 
> although dont waste your time unless youre bored and want something to do, my enjoyment from the game has died for the time being, the whole game needs an update, even with mods everything gets so repetitive 

yeah same, i haven't played it in a bit either because the slow updates. There's a lot to be improved with the NPC AI. but this type of stuff is my hobby in my free time as i love learning how to fix this stuff.

**Comment by Self-Replicating-Moron** [comment]
Yeah, the AI is janky af, i dont understand how they notice things. One second they dont care at all and cant see that youre picking a lock and next second someone a floor above you has seen you through the floor and pressed the emergency button, pulled a gun and started shooting me, then theres the mod that overhauls the system and suddenly nobody cares at all if youre beating the snot out of their partner while invading their home but SO HELP ME if you pick up a dustbin and drop it quietly everyone in a 2 block area comes to see whats going on and gets mad at you. 

but oh nice, how does fixing the mod work? im not familiar with how github works too much, but are you able to update his listing or would you have to make your own copy and upload it yourself to modio?

**Comment by ErraticFox** [comment]
> Yeah, the AI is janky af, i dont understand how they notice things. One second they dont care at all and cant see that youre picking a lock and next second someone a floor above you has seen you through the floor and pressed the emergency button, pulled a gun and started shooting me, then theres the mod that overhauls the system and suddenly nobody cares at all if youre beating the snot out of their partner while invading their home but SO HELP ME if you pick up a dustbin and drop it quietly everyone in a 2 block area comes to see whats going on and gets mad at you.
> 
> but oh nice, how does fixing the mod work? im not familiar with how github works too much, but are you able to update his listing or would you have to make your own copy and upload it yourself to modio?

Well ususally people will make a fork of the repo (clone) and then add the modificition and push the update to the actual repo (the original, not clone) then the author can check over it and approve it. But since they've been AWAL for a bit, in this case people just fork the repo and then people use that one instead until the original one is able to be updated.

**Comment by Self-Replicating-Moron** [comment]
ah okay then, fingers crossed more modders come to the game, could use more of everything

**Comment by Arsonide** [comment]
This error is due to a bug in BepInEx, not Ventrix, and is harmless. I debugged it for a while, it causes no issues. It's possible that whatever issues you were having came from another mod.

**Comment by Self-Replicating-Moron** [comment]
Oh, huh, i tried it with other mods disabled, even solo and i kept getting the inv bug and stuff. Which mods have known conflicts?

I'll have to try it again but from what I remember there was something about it saving inv to a file but not reading that file on loading the world/game or something and a bunch of red erros with lots of strings of 0s in them,  yeh im an awful bug reporter lmao sorry, ill try to replicated it and send the errors when inv doesnt load, Ive only had the problem with ventrix sync discs installed though and removing it didnt fix the saves

also glad to see youre not dead! Love your mods

**Comment by ErraticFox** [comment]
> This error is due to a bug in BepInEx, not Ventrix, and is harmless. I debugged it for a while, it causes no issues. It's possible that whatever issues you were having came from another mod.

I mean it's been quite a bit since I experienced it, but I do believe @Self-Replicating-Moron is correct. I think I vaguely remember also trying it with just this mod enable and still was getting an error. I remember trying to get it to work for a bit with editing files too as I don't think I even installed many. 

I've spent a least around a hundred hours in total just debugging Skyrim and Rimworld mod lists so I'm no stranger to the 50/50 method of finding the culprit either. 😜

EDIT: BTW that isn't to discredit you or call you a liar, just backing up his claim. 😁

**Comment by Self-Replicating-Moron** [comment]
You know somethings not right when you havent played the game in a while and upon booting it it immediately freezes lmfao, i got more troubleshooting to do too disabling things

[Error  :     Unity] Coroutine continue failure 

We're off to a great start! lmao


Hopefully I can replicate the inv thing, need to play the world a bit though, fingers crossed

**Comment by Arsonide** [comment]
> I think I vaguely remember also trying it with just this mod enable and still was getting an error.

Yes, the error will print, but it doesn't have any side effects. The coroutine the error claims isn't working...is working, and at least with Ventrix in isolation the game still ran and initialized fine.

If I removed the coroutine, the error went away, if I added it back, the error came back. So I found which coroutine it was, and after a few nights of debugging I found it was a known issue in the version of BepInEx that Shadows of Doubt is using.

**Comment by Arsonide** [comment]
> also glad to see youre not dead! Love your mods

I have been busy with work, I hit Shadow of Doubt modding really hard a few months last year to make Babbler and Ventrix, and I needed a bit of a break...which was made a bit easier by the developers ceasing most development.

I'm not on a permanent hiatus, I still think Shadows of Doubt has a lot of unrealized potential that mods could realize. I just have a lot going on outside of modding right now.
